### PR TITLE
Enables cookie exclusion via TraceProperties.

### PR DESCRIPTION
**Pull request by venilnoronha**
I've added a `excludeCookies` property to `TraceProperties`, which on setting `true` excludes `Cookie` and `Set-Cookie` from request/response headers. See #5982 for more info.

- [X] I have signed the CLA

Please review and pull.

Thanks,
Venil Noronha

### CODE REVIEW

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/trace/TraceProperties.java [hunk 1]
```diff
@@ -48,6 +49,12 @@
 	 */
 	private Set<Include> include = new HashSet<Include>(DEFAULT_INCLUDES);
 
+	/**
+	 * Whether {@literal Cookie} and {@literal Set-Cookie} should be excluded from
+	 * request/response headers. Defaults to {@code false}.
+	 */
+	private Boolean excludeCookies = false;
```
Comment (snicoll) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/trace/TraceProperties.java]: I think we should probably have an `Exclude` that pairs with the `Include` above. I don't know if having a bunch of well-known excludes is the way to go though. I have no better idea at this point.

Comment (venilnoronha) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/trace/TraceProperties.java]: Any thoughts on how we should map `Include`s with corresponding `Exlude`s in `TraceProperties`?

Comment (snicoll) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/trace/TraceProperties.java]: I am not sure. I think I'd have a `Exclude.COOKIE` that explains what that does and apply the excludes on top of the includes or something?

**Comment by venilnoronha** [comment]
Is the looking good now?

**Comment by snicoll** [comment]
It's obviously fixing the problem of having to define an exclude at all but I am not sure I like the idea of having to know all the default includes as soon as you want to deviate from that. We might have other areas in the framework when we do this though so I'll have a closer look today.

**Pull request closed without merging**